lib/LLVMIRCodeGen/GlowJIT.cpp

```diff
@@ -130,16 +130,35 @@ GlowJIT::GlowJIT(llvm::TargetMachine &TM)
                    }),
 #else
       SSP_(std::make_shared<SymbolStringPool>()), ES_(SSP_),
+      cxxoverride_([this](const std::string &S) {
+        std::string mangledName;
+        raw_string_ostream MangledNameStream(mangledName);
+        Mangler::getNameWithPrefix(MangledNameStream, S, DL_);
+        return MangledNameStream.str();
+      }),
       resolver_(createLegacyLookupResolver(
           ES_,
           [this](const std::string &Name) -> JITSymbol {
             if (auto Sym = compileLayer_.findSymbol(Name, false))
               return Sym;
             else if (auto Err = Sym.takeError())
               return std::move(Err);
+            // Some symbols are overriden, in particular __dso_handle and
+            // __cxa_atexit
+            if (auto overridenSym = cxxoverride_.searchOverrides(Name)) {
+              llvm::errs() << "JIT: Found overriden symbol '" << Name << "'\n";
+              return overridenSym;
+            }
+            // FIXME: looking for symbols external to libjit in the process is
+            // dangerous because it can be environment dependent. For example,
+            // we get cases where a symbol is found in the Linux environment,
+            // but not in the Windows environment.
             if (auto SymAddr =
                     RTDyldMemoryManager::getSymbolAddressInProcess(Name))
               return JITSymbol(SymAddr, JITSymbolFlags::Exported);
+            // The symbol was not resolved. Dump an error message and return a
+            // 'symbol not found' JITSymbol object (nullptr).
+            llvm::errs() << "JIT: Error resolving symbol '" << Name << "'\n";
```

Comment:
The problem is that when such case happens, it doesn't fail immediately. The error will happen later when trying to get the address of the 'jitmain' symbol (in LLVMCompiledFunction.cpp). We then get the "Error getting address" that doesn't tell us anything about what is the actual problem.

Reply:
I enclosed this into  `#ifndef NDEBUG`. I think it's important to keep it for the reason explained before. An we don't want to fail immediately because multiple symbols may not be resolved, so we want to capture all in one shot.

Reply:
@tlepley-cadence I'd suggest you just put this inside `DEBUG_GLOW`. This way you can always enable it via a command-line option.

Reply:
@tlepley-cadence Ping on this one. Please use `DEBUG_GLOW` instead of ifdefs

Reply:
Done.